application/src/main/java/org/togetherjava/tjbot/commands/moderation/UnbanCommand.java

```diff
@@ -0,0 +1,96 @@
+package org.togetherjava.tjbot.commands.moderation;
+
+import net.dv8tion.jda.api.Permission;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.Member;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
+import net.dv8tion.jda.api.exceptions.ErrorResponseException;
+import net.dv8tion.jda.api.interactions.Interaction;
+import net.dv8tion.jda.api.interactions.commands.OptionType;
+import net.dv8tion.jda.api.requests.ErrorResponse;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.tjbot.commands.SlashCommandAdapter;
+import org.togetherjava.tjbot.commands.SlashCommandVisibility;
+
+import java.util.Objects;
+
+/**
+ * Unbans a given user. Unbanning can also be paired with a reason. The command fails if the user is
+ * currently not banned.
+ */
+public final class UnbanCommand extends SlashCommandAdapter {
+    private static final Logger logger = LoggerFactory.getLogger(UnbanCommand.class);
+    private static final String TARGET_OPTION = "user";
+    private static final String REASON_OPTION = "reason";
+
+    /**
+     * Constructs an instance.
+     */
```

Comment:
```suggestion
```

Reply:
Public members must have javadoc. Its unfortunate that we can not configure it to exclude no-arg constructors or similar.